src/EventStore.Core.Tests/Http/Cluster/when_requesting_from_follower.cs

```diff
@@ -14,32 +14,43 @@ namespace EventStore.Core.Tests.Http.Cluster {
 	[TestFixture(typeof(LogFormat.V2), typeof(string))]
 	[TestFixture(typeof(LogFormat.V3), typeof(uint))]
 	public class when_requesting_from_follower<TLogFormat, TStreamId> : specification_with_cluster<TLogFormat, TStreamId> {
-		const int Retries = 5;
 		private const string TestStream = "test-stream";
+		private const string TestDeleteStream = "test-stream-to-be-deleted";
 		private IPEndPoint _followerEndPoint;
 		private IPEndPoint _leaderEndPoint;
 		private HttpClient _client;
 
 		protected override async Task Given() {
 			var leader = GetLeader();
 			_leaderEndPoint = leader.HttpEndPoint;
+			var follower = GetFollowers().First();
+			_followerEndPoint = follower.HttpEndPoint;
+			_client = follower.CreateHttpClient();
 			
 			// Wait for the admin user to be created
 			await leader.AdminUserCreated;
-			var replica = GetFollowers().First();
-			_followerEndPoint = replica.HttpEndPoint;
-			_client = replica.CreateHttpClient();
 			// Wait for the admin user created event to be replicated before starting our tests
-			var leaderIndex = GetLeader().Db.Config.IndexCheckpoint.Read();
-			AssertEx.IsOrBecomesTrue(()=> replica.Db.Config.IndexCheckpoint.Read() >= leaderIndex, 
-				msg: $"Waiting for replica to reach index checkpoint timed out! ({leaderIndex})");
+			var leaderIndex = leader.Db.Config.IndexCheckpoint.Read();
+			AssertEx.IsOrBecomesTrue(()=> follower.Db.Config.IndexCheckpoint.Read() >= leaderIndex, 
+				timeout: TimeSpan.FromSeconds(10),
+				msg: $"Waiting for follower to reach index checkpoint timed out! ({leaderIndex})");
 
-			var path = $"streams/{TestStream}";
-			var response = await PostEvent(_followerEndPoint, path, requireLeader: false);
+			leaderIndex = leader.Db.Config.IndexCheckpoint.Read();
+			
+			var response = await PostEvent(_followerEndPoint, $"streams/{TestDeleteStream}", requireLeader: false);
+			Assert.AreEqual(HttpStatusCode.Created, response.StatusCode);
+			
+			response = await PostEvent(_followerEndPoint, $"streams/{TestStream}", requireLeader: false);
 			Assert.AreEqual(HttpStatusCode.Created, response.StatusCode);
-			leaderIndex = GetLeader().Db.Config.IndexCheckpoint.Read();
-			AssertEx.IsOrBecomesTrue(()=> replica.Db.Config.IndexCheckpoint.Read() >= leaderIndex,
-				msg: $"Waiting for test stream to be synced with follower timed out! ({leaderIndex})");
+
+			AssertEx.IsOrBecomesTrue(() => leader.Db.Config.IndexCheckpoint.Read() > leaderIndex,
```

Comment:
i think we're getting the leader index before appending events to two different streams and then waiting for the index checkpoint to move - but perhaps it could move for the first event and still the second hasn't been processed yet?

if thats right it might be useful to have a helper method that add and waits, and just call that method twice

Reply:
Good point, added it.